src/Module/Procedures/ProcedureRoute.php

```diff
@@ -68,53 +75,52 @@ public function match(Context $context): bool
     return false;
   }
 
+  /** @return $this */
   public function getHandler()
   {
     return $this;
   }
 
   /**
-   * @throws InvalidPayloadException
+   * @throws InvalidPayloadException|ReflectionException|Exception
    */
   public function handle(Context $c): Response
   {
     $procedure = new $this->_procedureClass();
     if(!($procedure instanceof Procedure))
     {
-      throw new \Exception("Invalid procedure class given");
+      throw new Exception("Invalid procedure class given");
     }
 
     if($procedure instanceof ContextAware)
     {
       $procedure->setContext($c);
     }
 
-    if(method_exists($procedure, 'execute'))
+    if(!method_exists($procedure, 'execute'))
     {
-      $plClass = $this->_endpoint->getPayloadClass();
-      if($plClass !== null)
-      {
-        $payload = new $plClass();
-        if($payload instanceof AbstractPayload)
-        {
-          $payload->fromContext($c);
-        }
+      throw new Exception("Unable to handle procedure: execute missing");
+    }
 
-        if(!$payload->isValid())
-        {
-          throw InvalidPayloadException::withType((new \ReflectionClass($payload))->getShortName());
-        }
+    $plClass = $this->_endpoint->getPayloadClass();
 
-        //Execute with payload
-        $response = $procedure->execute($payload);
-      }
-      else
-      {
-        //Execute without payload
-        $response = $procedure->execute();
-      }
-      return JsonResponse::create($response);
+    if (!$plClass)
+    {
+      return JsonResponse::create($procedure->execute());
```

Comment:
We should avoid creating the JsonResponse twice here.

Its preferred to create the final response in a single line for improved refactoring or modification in the future.

Another option to would be to have a method to convert the response to the JsonResponse.  This would give us the option to return as text/xml for different clients, or endpoints.

Reply:
Having an else statement increases code complexity. An early return does look like the better option here.

If we need to modify any of this logic we can override the method.